comms/dht/src/connectivity/mod.rs

```diff
@@ -853,13 +853,47 @@ impl DhtConnectivity {
         }
     }
 
+    async fn remove_unmanaged_peers_from_pools(&mut self) -> Result<(), DhtConnectivityError> {
+        self.remove_allow_list_peers_from_pools().await?;
+        self.remove_exlcuded_peers_from_pools().await
+    }
+
     async fn remove_allow_list_peers_from_pools(&mut self) -> Result<(), DhtConnectivityError> {
         let allow_list = self.peer_allow_list().await?;
         self.neighbours.retain(|n| !allow_list.contains(n));
         self.random_pool.retain(|n| !allow_list.contains(n));
         Ok(())
     }
 
+    async fn remove_exlcuded_peers_from_pools(&mut self) -> Result<(), DhtConnectivityError> {
+        if !self.config.excluded_dial_addresses.is_empty() {
+            let mut neighbours = Vec::with_capacity(self.neighbours.len());
+            for peer in &self.neighbours {
+                let addresses = self.peer_manager.get_peer_multi_addresses(peer).await?;
+                if !addresses
+                    .iter()
+                    .any(|addr| self.config.excluded_dial_addresses.contains(addr.address()))
+                {
+                    neighbours.push(peer.clone());
+                }
```

Comment:
```suggestion
                let addresses = self.peer_manager.get_peer_multi_addresses(peer).await?;
                if !addresses
                    .iter()
                    .all(|addr| self.config.excluded_dial_addresses.contains(addr.address()))
                {
                    neighbours.push(peer.clone());
                }
```
Should this not be all?
I don't think we don't want to dial a peer, we only don't want to dial an address. 
If the peer has an address we can dial that's not excluded we should use that

Reply:
Correct, I intended that but made a logical error here